app/components/modals/WithdrawModal.tsx

```diff
@@ -0,0 +1,267 @@
+'use client';
+
+import { useEffect, useState, useCallback } from 'react';
+
+interface WithdrawModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onComplete: () => void;
+  balance: number;
+  btcAddress: string;
+  lightningToken: string;
+}
+
+interface WithdrawQuote {
+  quantity: number;
+  fee: number;
+}
+
+type ModalState = 'loading' | 'quote' | 'processing' | 'success' | 'error';
+
+export default function WithdrawModal({
+  isOpen,
+  onClose,
+  onComplete,
+  balance,
+  btcAddress,
+  lightningToken
+}: WithdrawModalProps) {
+  const [modalState, setModalState] = useState<ModalState>('loading');
+  const [quote, setQuote] = useState<WithdrawQuote | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
```

Comment:
Declare `successTimerRef` for the race condition fix.

```suggestion
  const successTimerRef = useRef<NodeJS.Timeout | null>(null);

```